src/app/users/store/users.actions.ts

```diff
@@ -3,16 +3,21 @@ import { IUser } from '../user.interface';
 
 const begin = createAction('[Users] Get All - Begin');
 const success = createAction(
-  '[Users] Get All - Success',
+  '[Users Effect] Get All - Success',
   props<{ users: IUser[] }>()
 );
-const failure = createAction(
-  '[Users] Get All - Failure',
-  props<{ error: any }>()
+
+//delete user
+const deleteUserBegin = createAction(
+  '[Users] Delete - Begin',
+  props<{ id: number }>()
 );
 
+const deleteUserSuccess = createAction('[Users Effect] Delete - Success');
+
 export const getAllUsersActions = {
```

Comment:
this is very wrong. does not have ANY sense this way of organizing your actions.
export const getAllUsersActions   ->  OK you put the actions for getAll, why you also add there actions for delete ?
YOu just need to add a new export with the same pattern.  eg:

```typescript
export const getAllUsersActions = {
begin,
 success,
 failure
}

export const deleteUserActions = {
 begin,
 success,
 failure
}

export const createUserActions = {
 begin,
 success,
 failure
};

export const updateUserActions = {
 begin,
 success,
 failure
}

export const getUserActions = {
 begin,
 success,
 failure
}

export const updateUserActions = {
 begin,
 success,
 failure
}

```

Reply:
done thanks!!!

Reply:
I cannot name begin, success and failure more than once.. How can I solve this?

Reply:
IF you are doing it all on the same file (that is not the idea as I mentioned) you can fix that easy like this.

````
export const updateUserActions = {
 begin: updateBeging,
 success: updateSuccess,
 failure: updateFailure
}```


export const updateUserActions = {
 begin,
 success,
 failure
}

Reply:
Ok I did it in a similar way, before reading this message, please let me know
Thanks